File: src/main/java/META-INF/resources/dataforms/dialog/image/ImageDialog.js
/**
 * @fileOverview {@link ImageDialog}クラスを記述したファイルです。
 */

'use strict';

import { Dialog } from '../../controller/Dialog.js';

/**
 * @class ImageDialog 画像表示ダイアログクラス。
 *
 * @extends Dialog
 */
export class ImageDialog extends Dialog {
	/**
	 * HTMLエレメントとの対応付けを行います。
	 */
	attach() {
		super.attach();
	}

	/**
	 * 閉じる際に画像をクリアします。
	 */
	close() {
		let imgfld = this.getComponent("imageForm").getComponent("image");
		imgfld.setValue(null);
		super.close();
	}
}

File: src/main/java/jp/dataforms/fw/dialog/image/ImageDialog.java
package jp.dataforms.fw.dialog.image;

import jp.dataforms.fw.controller.Dialog;

/**
 * 画像表示ダイアログ。
 *
 */
public class ImageDialog extends Dialog {
	/**
	 * ダイアログ。
	 */
	public ImageDialog() {
		super("imageDialog");
		this.addForm(new ImageForm());
	}
}
